tools/tpm2_tool.c

```diff
@@ -149,22 +149,71 @@ int main(int argc, char **argv) {
      */
     umask(0117);
 
-    if (!strcmp(argv[0], "tpm2")) {
-        if (argc == 1 || (argc == 2 && (!strcmp(argv[1],"--help") ||
-            !strcmp(argv[1], "-h") || !strcmp(argv[1], "--help=man")))) {
-            char *options[2] = {"tpm2","--help=man"};
-            tpm2_handle_options(2, options, 0, 0, 0);
-            exit(tool_rc_success);
-        }
+    bool is_str_tpm2 = (strcmp(argv[0], "tpm2") == 0);
 
-        if ((argc == 2 && (!strcmp(argv[1],"--version") ||
-        !strcmp(argv[1], "-v") ))) {
-            tpm2_handle_options(argc, argv, 0, 0, 0);
-            exit(tool_rc_success);
-        }
+    bool is_no_opts = (argc == 1);
+
+    bool is_one_opt_specified = (argc == 2 && is_str_tpm2);
+
+    bool is_opt_str_help = (is_one_opt_specified &&
+        ((strcmp(argv[1],"--help") == 0) || (strcmp(argv[1],"-h") == 0) ||
+         (strcmp(argv[1],"--help=man") == 0)));
+
+    if (is_no_opts || is_opt_str_help) {
+        char *options[2] = {"tpm2","--help=man"};
+        tpm2_handle_options(2, options, 0, 0, 0);
+        exit(tool_rc_success);
+    }
 
+    bool is_opt_str_help_no_man = (is_one_opt_specified &&
+        (strcmp(argv[1],"--help=no-man") == 0));
+
+    if (is_opt_str_help_no_man) {
+        tpm2_tool_output("Specify [ -v | --version ], [-h | --help] or one of "
+            "the following tool name with corresponding tool options:\n"
+            "[ certifyX509certutil ]\n[ checkquote ]\n[ eventlog ]\n[ print ]"
```

Comment:
This list seems like it will be difficult to maintain; there is a run-time generated list in `tpm2_tools.c` that could be traversed to generate it instead.

Reply:
@osresearch, thanks I will take a look at that. Why is the symlink test failing though?